mstickerlib/src/tg/sticker.rs

```diff
@@ -89,11 +132,36 @@ impl Sticker {
 		);
 
 		// download sticker from telegram
-		let mut image = self.download_image(tg_config).await?;
-		// convert sticker from lottie to gif if neccessary
-		if let Some(format) = animation_format {
-			image = image.convert_tgs(format.to_owned()).await?;
-		}
+		let image = self
+			.image
+			.import(
+				animation_format,
+				database,
+				tg_config,
+				matrix_config,
+				&self.pack_name,
+				self.positon,
+				self.emoji.as_deref(),
+				false
+			)
+			.await?;
+		let thumb = match self.thumbnail.as_ref() {
+			None => None, //async map is currently not supported by std
```

Comment:
I have two questions.

1. What does that comment mean?
2. Do we want to skip adding a thumbnail just because telegram does not have one? Or do we instead want to downsize the sticker ourselves?

    Especially for lottie files, I can imagine that telegram doesn't need a thumbnail, as lottie files are vector animations. But converting them to webp/gif makes them size dependent, so adding a thumbnail for matrix would be helpful.

Reply:
1. `lf.thumbnail.as_ref().map(|f| f.foo().await?)` does not work.
2. I would like to add this later. I think this should be also impl for all sticker not only for telegram imports.